test/e2e/scalenodes.go

```diff
@@ -101,31 +104,52 @@ var _ = Describe("Scale nodes", func() {
 			err = waitForScale(mss.Items[1].Name)
 			Expect(err).NotTo(HaveOccurred())
 
+			replicas, err = countReplicas()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(replicas).To(BeEquivalentTo(minSupportedReplicas))
+
+			ms, err := clients.MachineAPI.MachineV1beta1().MachineSets(machineSetsNamespace).List(ctx, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(mss.Items).NotTo(BeEmpty())
+
+			// Find which machineset is still zeroed out, and re-add the replica
+			for _, machineset := range ms.Items {
+				if machineset.Status.Replicas != 1 {
+					err = scale(machineset.Name, 1)
```

Comment:
This is a bit brittle, relies on the starting state of 1 replica per machineset.  Prefer we use the original value and not assume something about the shape of the cluster.  Could stash the index of the MS that is scaled down and the original number of replicas.  Then don't need to loop, just set the replicas and wait for scaling.